### PR TITLE
Crud implementation: Add and remove elements

### PR DESCRIPTION
**Pull request by Maclenn77**
Hi, Microverse reviewers!

I've applied the requested changes for this project.

- I removed all hardcoded items from the tasks array.
- I create a new JavaScript file for the new functionalities. I called **crud.js**.
- I implemented a function for adding a new task (add a new element to the array). (**createTask(a, b)**)
- I implemented a function for editing task descriptions (it's called **update(a, b, c)**). When you click the text, the color changes and the content is editable. Changes are saved when you press enter.
- I implemented a function for deleting a task (remove an element from the array, it's called **removeTaskFrom(a, b)**).
- I implement a function for the "Clear all completed" button (**removeAllFrom(a)**. It uses filter method as suggested).
- By default, new tasks have the property completed set to false and the property index set to the value of the new array length. I achieved this by creating a class for Task that sets the values from the beginning.
- Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique. 
- All changes to the To Do List are saved in local storage.

Have a nice day!
JP

### CODE REVIEW

File: src/crud.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// File with CRUD methods
```
Comment (SajjadAhmad14) [inline src/crud.js:1]: The project is awesome! But the user is not able to edit an item on the task list. Please add a function that enables the user to edit the task list description.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
GitHub flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/crud.js (hunk 1 above)
Comment (Maclenn77) [reply src/crud.js:1]: Hi, @SajjadAhmad14 !

Thanks for your review and comments. In this version, it is already available to edit the text. When you click on the text, it turns yellow and users can modify the description. The changes are saved after clicking enter. 

Maybe the solution is not very intuitive. I'll add instructions in the readme file and try to figure out how to do the editing feature more intuitive.

**Review by dasileker — APPROVED** [review]
## **Hi @Maclenn77,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**